app/src/androidTest/java/com/mutualmobile/barricade/sample/MainActivityTest.java

```diff
@@ -0,0 +1,60 @@
+package com.mutualmobile.barricade.sample;
+
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.IdlingResource;
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import com.jakewharton.espresso.OkHttp3IdlingResource;
+import com.mutualmobile.barricade.Barricade;
+import com.mutualmobile.barricade.BarricadeConfig;
+import com.mutualmobile.barricade.sample.api.util.ApiUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.not;
+
+/**
+ * Contains Espresso tests for the UI. Using Barricade, we can set a response for an endpoint, exercise the API call and verify the UI behavior.
+ */
+@RunWith(AndroidJUnit4.class) @LargeTest public class MainActivityTest {
+
+  @Rule public ActivityTestRule<MainActivity> mActivityRule = new ActivityTestRule<>(MainActivity.class);
+
+  private static final String JOKE_FROM_BARRICADE = "Gordon Ramsay features Chuck Norris' Toaster Strudel recipe in all of his restaurants.";
+
+  @Before public void registerIdlingResource() {
+    IdlingResource resource = OkHttp3IdlingResource.create("OkHttp", ApiUtils.getOkHttpClient());
+    Espresso.registerIdlingResources(resource);
+  }
+
+  @After public void teardown() {
+    Barricade.getInstance().reset();
+  }
+
+  @Test public void verifyEnablingBarricade() {
+    onView(withId(R.id.barricade_switch)).perform(click());
+    onView(withId(R.id.get_joke_button)).perform(click());
+    onView(withId(R.id.joke_text)).check(matches(withText(JOKE_FROM_BARRICADE)));
+  }
+
+  @Test public void verifyDisablingBarricade() {
+    onView(withId(R.id.get_joke_button)).perform(click());
+    onView(withId(R.id.joke_text)).check(matches(not(withText(JOKE_FROM_BARRICADE))));
```

Comment:
It is possible that we get the exact same joke from backend too. Only solution on the top of my head is to append a unique string like "- Barricade" to the joke returned by barricade. @mustafa01ali what do you think?